packages/api/src/services/popular_reads.ts

```diff
@@ -88,6 +95,51 @@ export const addPopularRead = async (
   return pageId
 }
 
+const addPopularReads = async (
+  userId: string,
+  names: string[]
+): Promise<AddPopularReadResult[]> => {
+  const results: AddPopularReadResult[] = []
+  for (const name of names) {
+    const pageId = await addPopularRead(userId, name)
+    results.push({
+      pageId,
+      name,
+      status: pageId
+        ? ArticleSavingRequestStatus.Succeeded
+        : ArticleSavingRequestStatus.Failed,
+    })
+  }
+  return results
```

Comment:
nice, could `addPopularRead` throw? I guess if one fails all probably will anyways.

Reply:
We handler the exception when creating pages in elastic so it should not throw any error but the `pageId` could be `undefined` if the `name` is not found in the array of `popularReads`, so it will add a failed result in the array of results